olm/addDigests.sh

```diff
@@ -50,32 +49,64 @@ rm -Rf ${BASE_DIR}/generated/${CSV_NAME}/
 mkdir -p ${BASE_DIR}/generated/${CSV_NAME}/
 cp -R ${BASE_DIR}/${SRC_DIR}/* ${BASE_DIR}/generated/${CSV_NAME}/
 
-CSV_FILE="$(find ${BASE_DIR}/generated/${CSV_NAME}/*${VERSION}/ -name "${CSV_NAME}.*${VERSION}.clusterserviceversion.yaml" | tail -1)"; # echo "[INFO] CSV = ${CSV_FILE}"
-${SCRIPTS_DIR}/buildDigestMap.sh -w ${BASE_DIR} -c ${CSV_FILE} -v ${VERSION} ${QUIET}
+CSV_FILE="$(find ${BASE_DIR}/generated/${CSV_NAME}/*${VERSION}/ -name "${CSV_NAME}.*${VERSION}.clusterserviceversion.yaml" | tail -1)"; 
 
-# inject relatedImages block
-names=" "
-count=1
-RELATED_IMAGES='. * { spec : { relatedImages: [ '
+echo "[INFO] CSV file path = ${CSV_FILE}"
+
+${BASE_DIR}/buildDigestMap.sh -w ${BASE_DIR} -c ${CSV_FILE} -v ${VERSION} ${QUIET}
+
+# inject relatedImages to CSV
 if [[ ! "${QUIET}" ]]; then cat ${BASE_DIR}/generated/digests-mapping.txt; fi
-for mapping in $(cat ${BASE_DIR}/generated/digests-mapping.txt)
+
+CSV_FILE_IN_MEMORY=$(cat ${CSV_FILE})
+DEFAULT_IMAGES=$( yq -r '.spec.install.spec.deployments[0].spec.template.spec.containers[0].env[] |
+                select(.name | startswith("IMAGE_default")) | 
+                [.value]' "${CSV_FILE}" | \
+                # Convert default images list to yaml/json "array"
+                jq -s 'reduce .[] as $item ([]; . + $item)')
+OPERATOR_IMAGE=$( yq -r '.spec.install.spec.deployments[0].spec.template.spec.containers[0].image' "${CSV_FILE}")
+REQUIRED_IMAGES=$( echo "${DEFAULT_IMAGES}" | yq -r ". += [\"${OPERATOR_IMAGE}\"]")
+RELATED_IMAGE_PREFIX="RELATED_IMAGE_"
+
+echo "[INFO] Generate digest update for CSV file..."
+
+for mapping in $(cat "${BASE_DIR}/generated/digests-mapping.txt")
 do
   source=$(echo "${mapping}" | sed -e 's/\(.*\)=.*/\1/')
   dest=$(echo "${mapping}" | sed -e 's/.*=\(.*\)/\1/')
-  sed -i -e "s;${source};${dest};" ${CSV_FILE}
-  name=$(echo "${dest}" | sed -e 's;.*/\([^\/][^\/]*\)@.*;\1;')
-  nameWithSpaces=" ${name} "
-  if [[ "${names}" != *${nameWithSpaces}* ]]; then
-    if [ "${names}" != " " ]; then
-      RELATED_IMAGES="${RELATED_IMAGES},"
-    fi
-    RELATED_IMAGES="${RELATED_IMAGES} { name: \"${name}\", image: \"${dest}\", tag: \"${source}\"}"
-    names="${names} ${name} "
-  fi
+  name=$(echo "${source}" | sed -e 's;.*/\([^\/][^\/]*\);\1;' | sed -r 's/[:]+/-/g')
+
+  nameForEnv=$(echo "${name}" | sed -r 's/[:\.\@-]+/_/g')
+  relatedImageEnvName="${RELATED_IMAGE_PREFIX}${nameForEnv}"
+
+CSV_FILE_IN_MEMORY=$( echo "${CSV_FILE_IN_MEMORY}" | \
```

Comment:
this new section of code seems much more complicated than what we're doing for CRW.

https://github.com/redhat-developer/codeready-workspaces-operator/tree/master/build/scripts

Why does it need to be some complicated?

Reply:
Because we have an issue 
https://github.com/eclipse/che/issues/14993 with definitions of done:

> 1) Add a relatedImages section in the ClusterServiceVersionSpec. Some images should be flagged as required (che-server, keycloak, postres, devfile and plugin registry, che-theia, brokers etc...) others not (plugins sidecars and stacks)
> 2) Update the operator deployment yaml and the operator itself to leverage the olm.relatedImage annotations

To satisfy "definition of done" I followed spec: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/contributors/design-proposals/related-images.md

Reply:
that spec is listed as "Status: Pending" and "Version: Alpha". So... I would make sure it's final before implementing it.

Meanwhile we already have relatedImages in use in CRW operator.

https://github.com/redhat-developer/codeready-workspaces-operator/blob/master/controller-manifests/v2.1.1/codeready-workspaces.csv.yaml#L421

That only uses two fields: `name` and `image`. I don't know if the other stuff like `annotation` or `required` are implemented in OCP yet.

Reply:
Yes, you are right. I reverted some changes related to the unimplemented spec... Related comment https://github.com/eclipse/che/issues/14993#issuecomment-637540933